### ISSUE TITLE
Improve documentation for replace and replace_command

### ISSUE THREAD

**Issue by Cyclonit** [orhun/git-cliff#609]
### Is there an existing issue or pull request for this?

- [X] I have searched the existing issues and pull requests

### Feature description

`changelog.postprocessors` and `git.commit_preprocessors` work on the entire changelog or git message respectively. The documentation does not clearly describe this. As a first time user, one might expect that either processor only affects the part of the input that actually matches the `pattern`.

For example, a postprocessor like the following would be expected to replace all occurrences of `<REPO>` with `https://github.com/orhun/git-cliff`. It replaces the entire changelog instead.

```toml
postprocessors = [
  { pattern = '<REPO>', replace_template = 'https://github.com/orhun/git-cliff' }
]
```

### Desired solution

The documentation should be revised in all places to clearly state that `pattern` does not limit the scope of the replacement, but instead is only used to check if a given string is supposed to replaced entirely.

### Alternatives considered

None

**Comment by orhun** [comment]
Agreed. We can also mention something like:

```
{ pattern = '.*', replace_command = 'typos --write-changes -' }
```

is the equivalent of:

```sh
echo "<matched_part_of_the_changelog>" | typos --write-changes -
```

Marked as good first issue, feel free to tackle this as well!

**Comment by Muhammad-Owais-Warsi** [comment]
Hi @orhun, for some reason, I don't have clarity on the issue, but I would love to tackle this. Could you please provide more details? Thanks!

**Comment by Muhammad-Owais-Warsi** [comment]
So basically I have to make changes [here](https://git-cliff.org/docs/configuration/git) , as mentioned in the comment by you and @Cyclonit . In this command `https://git-cliff.org/docs/configuration/git` and under the `commit_preprocessors`, I have to mention abou the `changelog.postprocessors`

**Comment by orhun** [comment]
Yeah, at there and other relevant sections in the documentation. I would recommend trying out this feature for yourself to get a more clear idea about what to do here ;)

**Comment by Muhammad-Owais-Warsi** [comment]
Hey, I have made some little changes in the documentation addressing the issue mentioned above.

1. The [commit_preprocessors](https://git-cliff.org/docs/configuration/git#commit_preprocessors) section.

![Screenshot from 2024-10-19 14-56-52](https://github.com/user-attachments/assets/6b2db356-cd94-486a-97b3-11bcaa7568b8)

2. In the [commit_parsers](https://git-cliff.org/docs/configuration/git#commit_parsers) section.

![Screenshot from 2024-10-19 15-01-53](https://github.com/user-attachments/assets/823fc1eb-6d50-4ebc-a6cb-87da3792dbd0)

Let me know whether the changes are correct and if any other parts of the documentation need updates. Thanks!

**Comment by orhun** [comment]
We can use "commit message" instead of "commit log". Other than that it seems fine, please send a PR - it is easier to discuss on the changes :)

**Comment by Muhammad-Owais-Warsi** [comment]
Sure @orhun

**Comment by Muhammad-Owais-Warsi** [comment]
In my PR #928 this [test](https://github.com/orhun/git-cliff/actions/runs/11462538444/job/31894169188?pr=928) got failed.  I tried to ran `rustfmt`,  but no processing took place.

**Comment by Muhammad-Owais-Warsi** [comment]
> In my PR #928 this [test](https://github.com/orhun/git-cliff/actions/runs/11462538444/job/31894169188?pr=928) got failed. I tried to ran `rustfmt`, but no processing took place, but instead formatting test passed.

**Comment by orhun** [comment]
That's fine. Can also mention https://github.com/orhun/git-cliff/issues/609#issuecomment-2053730878 in your PR?

**Comment by Muhammad-Owais-Warsi** [comment]
> That's fine. Can also mention [#609 (comment)](https://github.com/orhun/git-cliff/issues/609#issuecomment-2053730878) in your PR?

![Screenshot from 2024-10-24 19-22-47](https://github.com/user-attachments/assets/bbc7c28c-28f4-4914-a1f5-7b499b5c7495)

**Comment by Muhammad-Owais-Warsi** [comment]
#928
